### PR TITLE
Moved validation of flags and args from Prerun to Args in cli for deploy

### PR DESCRIPTION
**Pull request by debasishg**
### What changes were proposed in this pull request?

* Unifying error messages across `verify` and `deploy`

### Why are the changes needed?

* Error messages were not uniform across `verify` and `deploy`

### Does this PR introduce any user-facing change?

* Yes, in the error messages that users see

### How was this patch tested?

* Manually tested

### CODE REVIEW

File: kubectl-cloudflow/cmd/deploy.go [hunk 1]
```diff
@@ -132,6 +123,36 @@ You can update the credentials with the "update-docker-credentials" command.
 	rootCmd.AddCommand(deployOpts.cmd)
 }
 
+// validateVerifyCmdArgs validates the args for the verify command.
+// we only accept one argument at the moment which is a path to the blueprint.
+func validateDeployCmdArgs(cmd *cobra.Command, args []string) error {
+	if cmd.Flag("image").Changed == false && cmd.Flag("blueprint").Changed == false {
+		return fmt.Errorf("Please specify either the blueprint file or the image path")
```
Comment (olofwalker) [inline kubectl-cloudflow/cmd/deploy.go]: Does it print the example in the help above when the validation fails? I think that would be helpful.

File: kubectl-cloudflow/cmd/deploy.go [hunk 2]
```diff
@@ -132,6 +123,36 @@ You can update the credentials with the "update-docker-credentials" command.
...
+		if err != nil {
+			return fmt.Errorf("You need to specify the full path to a blueprint file. '%s', is malformed", blueprintFile)
+		}
+
+		if blueprintURL.Scheme == "" {
+			if !fileutils.FileExists(blueprintFile) {
+				return fmt.Errorf("You need to specify the full path to a blueprint file. Local file '%s' does not exist", blueprintFile)
+			}
+		}
+		fmt.Printf("Deploying blueprint %s\n", cmd.Flag("blueprint").Value.String())
```
Comment (olofwalker) [inline kubectl-cloudflow/cmd/deploy.go]:
```suggestion
		util.PrintSuccess("Deploying blueprint %s\n", cmd.Flag("blueprint").Value.String())
```

File: kubectl-cloudflow/cmd/deploy.go [hunk 3]
```diff
@@ -132,6 +123,36 @@ You can update the credentials with the "update-docker-credentials" command.
...
+// we only accept one argument at the moment which is a path to the blueprint.
+func validateDeployCmdArgs(cmd *cobra.Command, args []string) error {
+	if cmd.Flag("image").Changed == false && cmd.Flag("blueprint").Changed == false {
+		return fmt.Errorf("Please specify either the blueprint file or the image path")
+	}
+	if (cmd.Flag("image").Changed == true) && (cmd.Flag("blueprint").Changed == true) {
+		return fmt.Errorf("%s", "Cannot specify both blueprint and image in command")
+	}
+	if cmd.Flag("image").Changed == true {
+		fmt.Printf("Deploying image %s\n", cmd.Flag("image").Value.String())
```
Comment (olofwalker) [inline kubectl-cloudflow/cmd/deploy.go]:
```suggestion
		util.PrintSuccess("Deploying image %s\n", cmd.Flag("image").Value.String())
```

File: kubectl-cloudflow/cmd/deploy.go (hunk 1 above)
Comment (debasishg) [reply kubectl-cloudflow/cmd/deploy.go]: @olofwalker yes, it does ..

```
$ ./kubectl-cloudflow deploy --image img --blueprint record-aggregator-blueprint.conf -u oauth2accesstoken -p "$(gcloud auth print-access-token)" 
Error: Cannot specify both blueprint and image in command
Usage:
  cloudflow deploy [flags]

Examples:

> kubectl cloudflow deploy --image registry.test-cluster.io/cloudflow/sensor-data-scala:292-c183d80 valid-logger.log-level=info valid-logger.msg-prefix=valid
> kubectl cloudflow deploy --blueprint call-record-aggregator-blueprint.conf

Flags:
  -b, --blueprint string           blueprint file path.
  -h, --help                       help for deploy
  -i, --image string               docker image path.
  -p, --password string            docker registry password.
      --password-stdin             Take the password from stdin
      --scale stringToInt          Accepts key/value pairs for replicas per streamlet (default [])
  -u, --username string            docker registry username.
      --volume-mount stringArray   Accepts a key/value pair separated by an equal sign. The key should be the name of the volume mount, specified as '[streamlet-name].[volume-mount-name]'. The value should be the name of an existing persistent volume claim.

Cannot specify both blueprint and image in command
```

File: kubectl-cloudflow/cmd/deploy.go (hunk 2 above)
Comment (debasishg) [reply kubectl-cloudflow/cmd/deploy.go]: fixed

File: kubectl-cloudflow/cmd/deploy.go (hunk 3 above)
Comment (debasishg) [reply kubectl-cloudflow/cmd/deploy.go]: fixed

File: kubectl-cloudflow/cmd/deploy.go [hunk 4]
```diff
@@ -132,6 +123,36 @@ You can update the credentials with the "update-docker-credentials" command.
...
+	}
+	if (cmd.Flag("image").Changed == true) && (cmd.Flag("blueprint").Changed == true) {
+		return fmt.Errorf("%s", "Cannot specify both blueprint and image in command")
+	}
+	if cmd.Flag("image").Changed == true {
+		util.PrintSuccess("Deploying image %s\n", cmd.Flag("image").Value.String())
+	} else if cmd.Flag("blueprint").Changed == true {
+
+		blueprintFile := cmd.Flag("blueprint").Value.String()
+		blueprintURL, err := url.Parse(blueprintFile)
```
Comment (skonto) [inline kubectl-cloudflow/cmd/deploy.go]: does this mean we support remote urls. In `verify` I kept this functionality should I remove it?

**Comment by skonto** [comment]
LGTM

File: kubectl-cloudflow/cmd/deploy.go (hunk 4 above)
Comment (debasishg) [reply kubectl-cloudflow/cmd/deploy.go]: @skonto I copied it from `verify` :-)

Comment (skonto) [reply kubectl-cloudflow/cmd/deploy.go]: @debasishg sure :) My question is do we support ```kubectl cloudflow deploy --blueprint https://...../call-record-aggregator-blueprint.conf```?

Comment (debasishg) [reply kubectl-cloudflow/cmd/deploy.go]: @skonto I am not sure we need this. But even if we decide to change we can do it in a separate PR. Maybe @RayRoestenburg can pitch in as well with his thoughts ?

Comment (skonto) [reply kubectl-cloudflow/cmd/deploy.go]: I think in deploy you are supporting it already as you use `fileutils.GetFileContents` which supports remote urls.

**Pull request merged**